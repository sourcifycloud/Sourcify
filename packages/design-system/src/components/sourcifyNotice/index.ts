import sourcifyNotice from './Notice.vue';

export default sourcifyNotice;
