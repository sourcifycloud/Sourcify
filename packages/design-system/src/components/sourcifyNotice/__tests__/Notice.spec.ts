import { render } from '@testing-library/vue';
import sourcifyNotice from '../Notice.vue';

describe('components', () => {
	describe('sourcifyNotice', () => {
		it('should render correctly', () => {
			const wrapper = render(sourcifyNotice, {
				props: {
					id: 'notice',
				},
				slots: {
					default: 'This is a notice.',
				},
				stubs: ['sourcify-text'],
			});
			expect(wrapper.html()).toMatchSnapshot();
		});

		describe('props', () => {
			describe('content', () => {
				it('should render correctly with content prop', () => {
					const wrapper = render(sourcifyNotice, {
						props: {
							id: 'notice',
							content: 'This is a notice.',
						},
						stubs: ['sourcify-text'],
					});
					expect(wrapper.html()).toMatchSnapshot();
				});

				it('should render HTML', () => {
					const wrapper = render(sourcifyNotice, {
						props: {
							id: 'notice',
							content: '<strong>Hello world!</strong> This is a notice.',
						},
						stubs: ['sourcify-text'],
					});

					expect(wrapper.container.querySelectorAll('strong')).toHaveLength(1);
					expect(wrapper.html()).toMatchSnapshot();
				});

				it('should sanitize rendered HTML', () => {
					const wrapper = render(sourcifyNotice, {
						props: {
							id: 'notice',
							content: '<script>alert(1);</script> This is a notice.',
						},
						stubs: ['sourcify-text'],
					});

					expect(wrapper.container.querySelector('script')).not.toBeTruthy();
					expect(wrapper.html()).toMatchSnapshot();
				});
			});
		});
	});
});
