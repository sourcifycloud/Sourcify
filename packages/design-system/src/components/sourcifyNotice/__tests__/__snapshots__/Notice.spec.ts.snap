// Vitest Snapshot v1

exports[`components > sourcifyNotice > props > content > should render HTML 1`] = `
"<div id=\\"notice\\" role=\\"alert\\" class=\\"notice notice warning\\">
  <div class=\\"notice-content\\">
    <sourcify-text-stub size=\\"small\\" compact=\\"true\\" tag=\\"span\\"><span id=\\"notice-content\\" role=\\"region\\" class=\\"truncated\\"><strong>Hello world!</strong> This is a notice.</span></sourcify-text-stub>
  </div>
</div>"
`;

exports[`components > sourcifyNotice > props > content > should render correctly with content prop 1`] = `
"<div id=\\"notice\\" role=\\"alert\\" class=\\"notice notice warning\\">
  <div class=\\"notice-content\\">
    <sourcify-text-stub size=\\"small\\" compact=\\"true\\" tag=\\"span\\"><span id=\\"notice-content\\" role=\\"region\\" class=\\"truncated\\">This is a notice.</span></sourcify-text-stub>
  </div>
</div>"
`;

exports[`components > sourcifyNotice > props > content > should sanitize rendered HTML 1`] = `
"<div id=\\"notice\\" role=\\"alert\\" class=\\"notice notice warning\\">
  <div class=\\"notice-content\\">
    <sourcify-text-stub size=\\"small\\" compact=\\"true\\" tag=\\"span\\"><span id=\\"notice-content\\" role=\\"region\\" class=\\"truncated\\"> This is a notice.</span></sourcify-text-stub>
  </div>
</div>"
`;

exports[`components > sourcifyNotice > should render correctly 1`] = `
"<div id=\\"notice\\" role=\\"alert\\" class=\\"notice notice warning\\">
  <div class=\\"notice-content\\">
    <sourcify-text-stub size=\\"small\\" compact=\\"true\\" tag=\\"span\\">This is a notice.</sourcify-text-stub>
  </div>
</div>"
`;
